File: mobile/android/app/src/main/java/com/cdbl/leavemanager/data/api/NotificationService.kt
package com.cdbl.leavemanager.data.api

import com.cdbl.leavemanager.data.model.NotificationsResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface NotificationService {
    @GET("notifications/latest")
    suspend fun getNotifications(
        @Header("Authorization") token: String,
        @Query("limit") limit: Int = 20,
        @Query("unreadOnly") unreadOnly: Boolean = false
    ): Response<NotificationsResponse>

    @POST("notifications/{id}/read")
    suspend fun markRead(
        @Header("Authorization") token: String,
        @Path("id") id: Int
    ): Response<Map<String, Any>>
}

File: mobile/ios/CDBLLeaveManager/CDBLLeaveManager/Core/Networking/NotificationService.swift
//
//  NotificationService.swift
//  CDBLLeaveManager
//
//  API service for notifications.
//

import Foundation
import Combine

actor NotificationService {
    static let shared = NotificationService()
    private let apiClient = APIClient.shared
    
    private init() {}
    
    // MARK: - Get Notifications
    
    func getNotifications(limit: Int = 20, unreadOnly: Bool = false) async throws -> NotificationListResponse {
        return try await apiClient.request(
            "notifications/latest?limit=\(limit)&unreadOnly=\(unreadOnly)",
            method: .get
        )
    }
    
    // MARK: - Mark as Read
    
    func markAsRead(id: Int) async throws {
        let _: EmptyResponse = try await apiClient.request(
            "notifications/\(id)/read",
            method: .post
        )
    }
    
    func markAllAsRead() async throws {
        let _: EmptyResponse = try await apiClient.request(
            "notifications/read-all",
            method: .post
        )
    }
    
    // MARK: - Update Settings
    
    func updateSettings(emailEnabled: Bool, pushEnabled: Bool) async throws {
        struct Request: Encodable {
            let emailEnabled: Bool
            let pushEnabled: Bool
        }
        let _: EmptyResponse = try await apiClient.request(
            "notifications/settings",
            method: .put,
            body: Request(emailEnabled: emailEnabled, pushEnabled: pushEnabled)
        )
    }
}

// MARK: - Empty Response

struct EmptyResponse: Decodable {}
